src/app/frontend/deploy/kdupload_controller.js

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Controller for the kd upload directive.
+ *
+ * @final
+ */
+export default class KdUploadController {
+  /** @ngInject */
+  constructor() {
+    /**
+     * It will be set by kdFileReaderDirective to the instance of an input element
```

Comment:
Use present tense in comments. Such comments are easier to read and always true. This is what this project uses and internet generally suggests (though, as always, there are disputes).